core/GlobalState.cc

```diff
@@ -1069,16 +1073,16 @@ SymbolRef GlobalState::enterMethodSymbol(Loc loc, SymbolRef owner, NameRef name)
     if (store.exists()) {
         ENFORCE((store.data(*this)->flags & flags) == flags, "existing symbol has wrong flags");
         counterInc("symbols.hit");
-        return store;
+        return store.asMethodRef();
```

Comment:
In these methods, if the flags match then it must be a methodref since the flags encode that information.